openBCIBoard.js

```diff
@@ -495,18 +556,94 @@ function OpenBCIFactory() {
      *          Select to connect (true) all channels' N inputs to SRB1. This effects all pins,
      *              and disconnects all N inputs from the ADC.
      * @returns {Promise} resolves if sent, rejects on bad input or no board
+     * @author AJ Keller (@pushtheworldllc)
      */
     OpenBCIBoard.prototype.channelSet = function(channelNumber,powerDown,gain,inputType,bias,srb2,srb1) {
         var arrayOfCommands = [];
         return new Promise((resolve,reject) => {
-            k.getChannelSetter(channelNumber,powerDown,gain,inputType,bias,srb2,srb1).then((arr) => {
-                arrayOfCommands = arr;
-                resolve(this.write(arrayOfCommands));
+            k.getChannelSetter(channelNumber,powerDown,gain,inputType,bias,srb2,srb1)
+                .then((arr,newChannelSettingObject) => {
+                    arrayOfCommands = arr;
+                    this.channelSettingsArray[channelNumber-1] = newChannelSettingObject;
```

Comment:
I wanted to make sure that user's think that setting channel 7 actually sets channel 7 and not channel 6.